aptos-move/block-executor/src/executor.rs

```diff
@@ -109,45 +111,52 @@ where
     }
 
     fn execute(
-        &self,
         version: Version,
         signature_verified_block: &[T],
         last_input_output: &TxnLastInputOutput<T::Key, E::Output, E::Error>,
         versioned_cache: &MVHashMap<T::Key, T::Value, X>,
         scheduler: &Scheduler,
         executor: &E,
-        base_view: &S,
+        sync_view: &mut LatestView<T, S, X>,
     ) -> SchedulerTask {
         let _timer = TASK_EXECUTE_SECONDS.start_timer();
         let (idx_to_execute, incarnation) = version;
         let txn = &signature_verified_block[idx_to_execute as usize];
 
-        let speculative_view = MVHashMapView::new(versioned_cache, scheduler);
-
         // VM execution.
-        let execute_result = executor.execute_transaction(
-            &LatestView::<T, S, X>::new_mv_view(base_view, &speculative_view, idx_to_execute),
-            txn,
-            idx_to_execute,
-            false,
-        );
-        let mut prev_modified_keys = last_input_output.modified_keys(idx_to_execute);
+        sync_view.update_txn_idx(idx_to_execute);
```

Comment:
What does this do? Why does this replace `speculative_view`?

Reply:
reverted back to creating new every time after Zekun's comments